modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+|`iconOnly` |(optional) if true, only displays icon
+|`noTooltip` |(optional) if true, tooltip won't be displayed
+|`className` |(optional) additional class name for the component
+|`popoverTitle` |(optional) title for popover
+|===
+
+[discrete]
+== `checkAccess`
+
+Provides information about user access to a given resource. It returns an object with resource access information.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resourceAttributes` |resource attributes for access review
+|`impersonate` |impersonation details
+|===
+
+[discrete]
+== `useAccessReview`
+
+Hook that provides information about user access to a given resource. It returns an array with `isAllowed` and `loading` values.
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resourceAttributes` |resource attributes for access review
+|`impersonate` |impersonation details
+|===
+
+[discrete]
+== `useResolvedExtensions`
+
+React hook for consuming Console extensions with resolved `CodeRef` properties. This hook accepts the same argument(s) as `useExtensions` hook and returns an adapted list of extension instances, resolving all code references within each extension's properties.Initially, the hook returns an empty array. Once the resolution is complete, the React component is re-rendered with the hook returning an adapted list of extensions.When the list of matching extensions changes, the resolution is restarted. The hook will continue to return the previous result until the resolution completes.The hook's result elements are guaranteed to be referentially stable across re-renders. It returns a tuple containing a list of adapted extension instances with resolved code references, a boolean flag indicating whether the resolution is complete, and a list of errors detected during the resolution.
+
+[source,ts]
+----
+const [navItemExtensions, navItemsResolved] = useResolvedExtensions<NavItem>(isNavItem);
+// process adapted extensions and render your component
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`typeGuards` |A list of callbacks that each accept a dynamic plugin
+extension as an argument and return a boolean flag indicating whether or
+not the extension meets desired type constraints
+|===
+
+[discrete]
+== `HorizontalNav`
+
+A component that creates a Navigation bar for a page.- Routing is handled as part of the component.- `console.tab/horizontalNav` can be used to add additional content to any horizontal nav.
+
+[source,ts]
+----
+const HomePage: React.FC = (props) => {
+    const page = {
+      href: '/home',
+      name: 'Home',
+      component: () => <>Home</>
+    }
+    return <HorizontalNav match={props.match} pages={[page]} />
+}
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resource` |The resource associated with this Navigation, an object of
+K8sResourceCommon type
+
+|`pages` |An array of page objects
+
+|`match` |match object provided by React Router
```

Comment:
Also, some of the tables have an Optional Y/N column and some use (optional) in the description. Long-term project to make this consistent?

Reply:
I don't have knowledge how this file gets generated as of now, but I can clearly see these table capitalization inconsistencies in the [openshift/console](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/docs/api.md) too. Maybe it would be suitable to create issues for openshift/openshift-docs and openshift/console that would deal with this problem.